src/rdb.c

```diff
@@ -1183,30 +1183,48 @@ int rdbSaveInfoAuxFields(rio *rdb, int rdbflags, rdbSaveInfo *rsi) {
 ssize_t rdbSaveSingleModuleAux(rio *rdb, int when, moduleType *mt) {
     /* Save a module-specific aux value. */
     RedisModuleIO io;
-    int retval = rdbSaveType(rdb, RDB_OPCODE_MODULE_AUX);
-    if (retval == -1) return -1;
     moduleInitIOContext(io,mt,rdb,NULL,-1);
+
+    rio aux_save_headers_rio;
+    rioInitWithBuffer(&aux_save_headers_rio, sdsempty());
```

Comment:
let's add a short comment here too as to why we need that buffer.
i.e. so that on aux_save2 we can avoid saving the headers if it didn't save anything.